src/app/AttributePathParams.h

```diff
@@ -81,10 +83,49 @@ struct AttributePathParams
         return true;
     }
 
-    ClusterId mClusterId     = kInvalidClusterId;   // uint32
-    AttributeId mAttributeId = kInvalidAttributeId; // uint32
-    EndpointId mEndpointId   = kInvalidEndpointId;  // uint16
-    ListIndex mListIndex     = kInvalidListIndex;   // uint16
+    bool IncludesAttributesInCluster(const DataVersionFilter & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsAttributePathIntersect(const ConcreteClusterPath & other) const
+    {
+        VerifyOrReturnError(HasWildcardEndpointId() || mEndpointId == other.mEndpointId, false);
+        VerifyOrReturnError(HasWildcardClusterId() || mClusterId == other.mClusterId, false);
+
+        return true;
+    }
+
+    bool IsWildcardRequest(const AttributePathParams & aOther) const
+    {
+        if (HasWildcardEndpointId())
+        {
+            return true;
```

Comment:
This is incorrect logic. Just because endpoint is wildcard doesn't mean we have 'inclusion':

ReadPath = EP* / C1
DataPath = EP2 / C2 / A1

Reply:
updated, thanks